source/jormungandr/jormungandr/scenarios/helper_classes/DirectPath.py

```diff
@@ -0,0 +1,83 @@
+import Future
+from jormungandr import utils
+from collections import namedtuple
+from jormungandr.street_network.street_network import DirectPathType
+
+DirectPathKey = namedtuple('DirectPathKey', ['mode', 'orig_uri', 'dest_uri', 'direct_path_type'])
+
+class DirectPath:
+    def __init__(self, instance, orig_obj, dest_obj, mode, fallback_extremity, request, direct_path_type):
+        self._instance = instance
```

Comment:
actually it is instance.... we use instance to dispatch the computation [here](https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/instance.py#L484)